### PR TITLE
✨ Mobile | Claim rewards in-person

### LINKED ISSUE #314: In app rewards QR codes for admin staff
Staff with the Admin role should have an additional menu item "Rewards admin" (or similar).

When they tap it, they get a carousel with all available rewards (like the non-admin Rewards menu). But it shows the QR code.

So if someone wants to claim a reward, they can show them the QR code to scan from within the app/

### LINKED ISSUE #858: ✨ Mobile | Claim reward in-person
Hi team

### Pain

Currently we have a workflow in the app for redeeming rewards to be sent to an address, but no way to claim a reward in-person (such as at a conference).

### Suggested Solution

Update the redemption workflow to ask if the user is at an event and allow to be claimed without capturing an address.

### Tasks

- [ ] Investigate
- [ ] Implement solution

Thanks!

### PR DESCRIPTION
**Pull request by zacharykeeping**
> 1. What triggered this change? (PBI link, Email Subject, conversation + reason, etc)

Closes #858
Closes #314

> 2. What was changed?

Allows for user's to redeem rewards in-person, which generates a QR code which they must then show to SSW staff. SSW staff can then scan the code to confirm the redemption and provide the user with the reward.

**Figure: Users now have the option to redeem in-person**

**Figure: Redeeming in-person will display a QR code which must be then scanned by SSW staff. The user can also cancel the redemption if they've changed their mind. (Note: points aren't deducted until the redemption is confirmed)**

**Figure: If the user has closed the window of a pending redemption, they can tap the QR button to bring it up again**

**Figure: When SSW staff scans the code it will then be confirmed and they can hand over the reward. Points get deducted at this point.**

> 3. Did you do pair or mob programming?

No

### CODE REVIEW

File: src/MobileUI/PopupPages/RedeemReward.xaml [hunk 1]
```diff
@@ -8,6 +8,7 @@
                  xmlns:toolkit="http://schemas.microsoft.com/dotnet/2022/maui/toolkit"
                  xmlns:addr="clr-namespace:SSW.Rewards.Shared.DTOs.AddressTypes;assembly=Shared"
                  xmlns:lottie="clr-namespace:SkiaSharp.Extended.UI.Controls;assembly=SkiaSharp.Extended.UI"
+                 xmlns:controls1="clr-namespace:ZXing.Net.Maui.Controls;assembly=ZXing.Net.MAUI.Controls"
```
Comment (matt-goldman) [inline src/MobileUI/PopupPages/RedeemReward.xaml:11]: Typo?
```suggestion
                 xmlns:controls="clr-namespace:ZXing.Net.Maui.Controls;assembly=ZXing.Net.MAUI.Controls"
```

File: src/MobileUI/PopupPages/RedeemReward.xaml [hunk 2]
```diff
@@ -164,6 +164,24 @@
...
                     </Grid>
+                    
+                    <!-- Pending redemption QR code -->
+                    <VerticalStackLayout IsVisible="{Binding IsQrCodeVisible}"
+                                         Spacing="10"
+                                         Margin="0,0,0,30">
+                        <Label Text="Please show this code to SSW staff to claim your reward:"
+                               HorizontalOptions="Center"
+                               HorizontalTextAlignment="Center"/>
+                        <controls1:BarcodeGeneratorView
```
Comment (matt-goldman) [inline src/MobileUI/PopupPages/RedeemReward.xaml:175]: Typo?
```suggestion
                        <controls:BarcodeGeneratorView
```

**Review by matt-goldman — APPROVED** [review]
LGTM

Not sure if `Controls1` is a typo or if there's already a `Controls` namespace in that file. If the latter, use something meaningful rather than `Controls1` (e.g. (`BarcodeControls`)

**Comment by AntPolkanov** [comment]
Also closes #314

File: src/Application/SSW.Rewards.Application.csproj [hunk 3]
```diff
@@ -26,7 +26,7 @@
 
     <ItemGroup>
```
Comment (AntPolkanov) [inline src/Application/SSW.Rewards.Application.csproj:27]: I don't this we need this `ItemGroup`. 
AFAIN, `Folder` is used when you need to generate classes and put them into a non-existing folder.
[Docs](https://learn.microsoft.com/en-us/visualstudio/msbuild/common-msbuild-project-items?view=vs-2022#folder)

Comment (AntPolkanov) [reply src/Application/SSW.Rewards.Application.csproj]: Fixed

**Review by AntPolkanov — APPROVED** [review]
Great work! 💪

**Pull request merged**